### PR TITLE
[plugins] Fix Pylint and PEP8 issues - part2

### PR DESCRIPTION
**Pull request by pponnuvel**
Continuation of #3530.

Likewise excludes [C0114](https://pylint.readthedocs.io/en/latest/user_guide/messages/convention/missing-module-docstring.html), [C0115](https://pylint.readthedocs.io/en/latest/user_guide/messages/convention/missing-class-docstring.html), and [C0209](https://pylint.pycqa.org/en/latest/user_guide/messages/convention/consider-using-f-string.html).

Goes from files starting with [f-m].

Added some inline comments to explain.

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [ ] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [ ] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?

### CODE REVIEW

File: sos/report/plugins/foreman.py
Comment (pponnuvel) [inline sos/report/plugins/foreman.py:13]: `pipes` has been deprecated.

File: sos/report/plugins/foreman.py
Comment (pponnuvel) [inline sos/report/plugins/foreman.py:184]: In this file, pylint was complaining that 'R0912 too-many-branches' and 'R0914 too-many-locals' - essentially the function being too big. So split it into 3.

File: sos/report/plugins/foreman_installer.py
Comment (pponnuvel) [inline sos/report/plugins/foreman_installer.py:92]: Addresses "W1404: Implicit string concatenation found in list (implicit-str-concat)"

File: sos/report/plugins/gluster.py
Comment (pponnuvel) [inline sos/report/plugins/gluster.py:12]: `string` isn't really used here. See the next comment.

File: sos/report/plugins/gluster.py
Comment (pponnuvel) [inline sos/report/plugins/gluster.py:43]: `string` module doesn't have a `count` method. Rather the string class has it.

With the previous code, this would actually be an infinite loop since `ret` will always be -1. I believe this never worked as intended and the plugin typically gets killed on timeout.

File: sos/report/plugins/gluster.py
Comment (pponnuvel) [inline sos/report/plugins/gluster.py:107]: `W1203: Use lazy % formatting in logging functions (logging-fstring-interpolation)`

So replaced with f-string here .

File: sos/report/plugins/kubernetes.py
Comment (pponnuvel) [inline sos/report/plugins/kubernetes.py:37]: To address 'R0912 too-many-branches' and 'R0914 too-many-locals' - essentially the function being too big.

Moved vars around to make the split easier and logical.

File: sos/report/plugins/monit.py
Comment (pponnuvel) [inline sos/report/plugins/monit.py:24]: Pylint doesn't like FIXME comments. This was added in 2016. If no one's affected yet, I thought we can ignore it until someone actually hits this issue.

The alternative is to create an issue and link it there and say 'it's tracked in #issue'. Let me know if this is preferable here.

File: sos/report/plugins/gluster.py
Comment (pponnuvel) [inline sos/report/plugins/gluster.py:107]: `W1203: Use lazy % formatting in logging functions (logging-fstring-interpolation)`

So replaced with f-string here.

**Comment by pponnuvel** [comment]
The foreman tests are usually skipped, not sure why it's triggered here (this wasn't seen few days back in this PR either). I did a different change (unrelated to the foreman plugins) yesterday and now these started failing!

**Comment by pmoravec** [comment]
Let ignore the foreman failed tests, they appear on any PR now. I raised https://github.com/sosreport/sos/issues/3542 for it.

File: sos/report/plugins/foreman_installer.py
Comment (pmoravec) [reply sos/report/plugins/foreman_installer.py:92]: There is missing coma :)

Currently - and your PR keeps that behaviour - we get `manifest.json`:

                        {
                            "specification": "/var/log/foreman-installer/satellite.log",
                            "files_copied": [
                                "var/log/foreman-installer/satellite.log"
                            ],
                            "tags": [
                                "foreman_satellite_logsatellite_installer_log"
                            ]
                        },
                        {
                            "specification": "/var/log/foreman-installer/capsule.log",
                            "files_copied": [
                                "var/log/foreman-installer/capsule.log"
                            ],
                            "tags": [
                                "capsule_logcapsule_installer_log"
                            ]
                        }
    ..
            "tag_summary": {
                "capsule_logcapsule_installer_log": [
                    "var/log/foreman-installer/capsule.log"
                ],
                "foreman_satellite_logsatellite_installer_log": [
                    "var/log/foreman-installer/satellite.log"
                ]
            }

while we should get rather:

                        {
                            "specification": "/var/log/foreman-installer/satellite.log",
                            "files_copied": [
                                "var/log/foreman-installer/satellite.log"
                            ],
                            "tags": [
                                "foreman_satellite_log",
                                "satellite_installer_log"
                            ]
                        },
                        {
                            "specification": "/var/log/foreman-installer/capsule.log",
                            "files_copied": [
                                "var/log/foreman-installer/capsule.log"
                            ],
                            "tags": [
                                "capsule_log",
                                "capsule_installer_log"
                            ]
                        }
    ..
            "tag_summary": {
                "capsule_installer_log": [
                    "var/log/foreman-installer/capsule.log"
                ],
                "capsule_log": [
                    "var/log/foreman-installer/capsule.log"
                ],
                "foreman_satellite_log": [
                    "var/log/foreman-installer/satellite.log"
                ],
                "satellite_installer_log": [
                    "var/log/foreman-installer/satellite.log"
                ]
            }

Git blame shows some change due to https://github.com/sosreport/sos/commit/77614068aef89ad13492f819beabbad07a53dd47 and https://github.com/sosreport/sos/commit/962960c5e15e7349067a920aca7c8eae157e67d7 prior that. I doubt the current tags are really used (or @ryan-blakley or @TurboTurtle  is aware of their usage?). So I would not hesitate to change them to something more meaningful.

I would suggest:

    self.add_file_tags({
            '/var/log/foreman-installer/satellite.log':
                ['foreman_installer_log', 'satellite_installer_log'],
            '/var/log/foreman-installer/capsule.log':
                ['foreman_installer_log', 'capsule_installer_log'],
    })

as both logs are foreman_installer ones, one for installing Satellite, one for installing Capsule.

**Comment by pponnuvel** [comment]
> Let ignore the foreman failed tests, they appear on any PR now. I raised #3542 for it.

Thank you! (I was going to try few other things - you saved my time :)

File: sos/report/plugins/iscsi.py
Comment (pmoravec) [inline sos/report/plugins/iscsi.py:64]: Missing `super().setup()`, as well as in `DebianIscsi`?

File: sos/report/plugins/kata_containers.py
Comment (pmoravec) [inline sos/report/plugins/kata_containers.py:25]: Ack, dead code. Nice!

File: sos/report/plugins/kdump.py
Comment (pmoravec) [inline sos/report/plugins/kdump.py:59]: `kdump = '/etc/kdump.conf'` as it was in (redundantly called) `path_join`?

Comment (pponnuvel) [reply sos/report/plugins/kdump.py:59]: Ack, will fix.

File: sos/report/plugins/foreman_installer.py
Comment (pponnuvel) [reply sos/report/plugins/foreman_installer.py:92]: Thanks :) I'll modify as you suggested.

File: sos/report/plugins/iscsi.py
Comment (pponnuvel) [reply sos/report/plugins/iscsi.py:64]: Since `setup` isn't doing anything, `super().setup()` isn't needed at all. So I'll remove this setup definition from both `RedHatIscsi` and `DebianIscsi` classes.

File: sos/report/plugins/kubernetes.py
Comment (pmoravec) [inline sos/report/plugins/kubernetes.py:133]: Technically, we can call this line just once before the `for` cycle and save some CPU cycles.

File: sos/report/plugins/monit.py
Comment (pmoravec) [reply sos/report/plugins/monit.py:24]: ACK to remove this. If nobody has complained in past 8 years, most probably nothing to fix.

**Review by pmoravec — APPROVED** [review]
ACK, optionally with the ` kube_get_cmd` assignement moved.

Nice work!

**Review by pmoravec — APPROVED** [review]

**Comment by TurboTurtle** [comment]
Marking for review this weekend.

**Review by TurboTurtle — APPROVED** [review]
LGTM, though I'd prefer if we could get some direct validation on a kubernetes environment since there is a lot of re-arranging going on I just want to be absolutely sure we're maintaining behavior.

I no longer have access to one. @sosreport/core anyone?

**Comment by pmoravec** [comment]
> LGTM, though I'd prefer if we could get some direct validation on a kubernetes environment since there is a lot of re-arranging going on I just want to be absolutely sure we're maintaining behavior.
> 
> I no longer have access to one. @sosreport/core anyone?

Not me.. :(

**Comment by pponnuvel** [comment]
@pmoravec @TurboTurtle No worries. I'll setup & test this in a Kubernetes environment. Hold this from merge until I come back :)

**Comment by pponnuvel** [comment]
[kubernetes-sosreports.zip](https://github.com/sosreport/sos/files/14472229/kubernetes-sosreports.zip)

@pmoravec @TurboTurtle I've uploaded a zip file here. It contains 2 sets of sosreports with sosreport 4.5.6 and sosreport 4.7.0 (on top of this branch) and each set from a kubernetes control plane and worker node. I did a quick check (file counts, couple random files checked on content, any errors in sos.log & ui.log) and seemed OK. If you've time, please a take a look as well. Thanks!

**Comment by TurboTurtle** [comment]
> [kubernetes-sosreports.zip](https://github.com/sosreport/sos/files/14472229/kubernetes-sosreports.zip)
> 
> @pmoravec @TurboTurtle I've uploaded a zip file here. It contains 2 sets of sosreports with sosreport 4.5.6 and sosreport 4.7.0 (on top of this branch) and each set from a kubernetes control plane and a worker node. I did a quick check (file counts, couple random files checked on content, any errors in sos.log & ui.log) and seemed OK. If you've time, please a take a look as well. Thanks!

The report from this branch looks like I'd expect a kubernetes collection to look like. Thanks for setting that up!

**Pull request merged**
